regex-syntax/src/hir/translate.rs

```diff
@@ -318,12 +318,9 @@ impl<'t, 'p> Visitor for TranslatorI<'t, 'p> {
                         ast.negated,
                         &mut cls,
                     )?;
-                    if cls.iter().next().is_none() {
-                        return Err(self.error(
-                            ast.span,
-                            ErrorKind::EmptyClassNotAllowed,
-                        ));
-                    }
+                    cls.iter().next().is_none().ok_or_else(|| {
+                        self.error(ast.span, ErrorKind::EmptyClassNotAllowed)
+                    })?;
```

Comment:
This also just reads weird to me. It feels much more natural as an `if` statement.